lbz/authz.py

```diff
@@ -25,106 +21,74 @@
 LIMITED_ALLOW = -1
 
 
-class Authorizer(metaclass=Singleton):
-    allow = {}
-    deny = {}
-    action = None
-    outcome = None
-    allowed_resource = None
-    denied_resource = None
-    expiration = None
-    iss = None
-
-    def __init__(self, resource=None):
-        self.resource = resource
-        self._permissions = NestedDict()
-
-    def __getitem__(self, y):
-        return self._permissions[y]
-
-    def __str__(self):
-        return json.dumps({self.resource: self._permissions})
+class Authorizer:
+    def __init__(self, auth_jwt: str, resource_name: str, permission_name: str):
+        self.outcome = DENY
+        self.allowed_resource = None
+        self.denied_resource = None
+        self.resource = resource_name
+        self.permission = permission_name
+        self.allow = {}
+        self.deny = {}
+        self._set_policy(auth_jwt)
 
     def __repr__(self):
-        return self.__str__()
-
-    def __contains__(self, *args, **kwargs):
-        return self._permissions.__contains__(*args, **kwargs)
-
-    def __len__(self):
-        return len(self._permissions)
-
-    def __iter__(self):
```

Comment:
why remove all above ?

Reply:
- class attributes were converted to instance attributes since the class is not a singleton anymore
- I removed the `_permissions` dict since it seemed unnecessary, most of these deleted functions were related to that dict so they were removed too